src/Engine/ProtoCore/Reflection/MemberInfoReflectionExtensions.cs

```diff
@@ -0,0 +1,208 @@
+                                    paramValues.Add(cs.Value.CreateInstanceFromObject(cs.ArgumentType));
+                                else
+                                    paramValues.Add(e);
+                            }
+
+                            continue;
+                        }
+                        paramValues.Add(item.Value.CreateInstanceFromObject(item.ArgumentType));
+                    }
+
+
+                    dynamic ctorParam = null;
+                    if (paramValues.Count > 1)
+                    {
+                        var array = paramValues.ToArray<object>();
+                        if (TryConvertToTypeArray(array, out Array newArray))
+                        {
+                            ctorParam = newArray;
+                        }
+
+                        else
+                        {
+                            ctorParam = array;
+                        }
+                    }
+                    else
+                    {
+                        ctorParam = paramValues.FirstOrDefault();
+                    }  
+
+                    var a = CreateInstance(type, ctorParam);
+                    if (a is null) continue;
+
+                    attributes.Add(a);
+                }
+                catch {}
+            }
+
+            return attributes.ToArray();
+        }
+
+        private static Attribute CreateInstance(System.Type type, object args)
+        {
+            var loadedAssembly = Assembly.LoadFrom(type.Assembly.Location);
+            var loadedType = loadedAssembly.GetType(type.FullName);
+
+            if (!loadedType.GetConstructors().Any())
+                return null;
+
+            if (args is null)
+                return (Attribute)Activator.CreateInstance(loadedType);
+
+            var attr = args is ICollection ?
+                (Attribute)Activator.CreateInstance(loadedType, (object[])args) :
+                (Attribute)Activator.CreateInstance(loadedType, args);
+
+            return attr;
+        }
+
+        private static object CreateInstanceFromObject(this object obj, System.Type type)
+        {
+            if (type.IsPrimitive || type.Name == typeof(string).Name)
+                return obj;
+
+            if (type.IsEnum)
+                return InstanceFromEnum(type, (int)obj);
+
+            var typedObj = CreateInstance(type, obj);
+            if (typedObj is null)
+                return obj;
+
+            return typedObj;
+        }
+
+        private static object InstanceFromEnum(System.Type type, int value)
+        {
+            var loadedAssembly = Assembly.LoadFrom(type.Assembly.Location);
+            var loadedType = loadedAssembly.GetType(type.FullName);
+            if (!loadedType.IsEnum) return null;
+
+            return Enum.ToObject(loadedType, value);
+        }
+
+        private static bool TryConvertToTypeArray(object[] array, out Array typeArray)
```

Comment:
have you seen any cases where this needed to be recursive? Ie array of array?

Reply:
@mjkkirschner i havent seen that no, only seen param arrays like `params string[]`

Reply:
yeah, that makes sense because this is just attributes right? It seems uncommon - I'm not sure any of our attributes are that complex yet.

Reply:
yep only attributes